src/components/BrandModelDropdowns.tsx

```diff
@@ -0,0 +1,56 @@
+import { useEffect } from "react";
+import { UseFormRegister, UseFormWatch } from "react-hook-form";
+import { useModelsLazyQuery } from "../generated/graphql";
+import { IFormInputs } from "../pages/CarForm";
+import Dropdown from './Dropdown';
+
+interface DropdownProps {
+  brands: {
+    id: number,
+    name: string,
+  }[] | undefined;
+  isErrorBrand: boolean;
+  isErrorModel: boolean;
+  register: UseFormRegister<IFormInputs>;
+  watch: UseFormWatch<IFormInputs>
+}
+
+const BrandModelDropdowns = ({brands, isErrorBrand,isErrorModel, register, watch}: DropdownProps) => {
+  
+  const [fetchModels, { data: modelsData }] = useModelsLazyQuery();
+  const selectedBrand = watch('brand');
+
+  useEffect(() => {
+    fetchModels({
+      variables: {
+        where: {
+          brand_id: {
+            _eq: selectedBrand ? Number(selectedBrand) : null
```

Comment:
I suggest to use an if statement to wrap the fetchModels because the way you are using it the query can be called with a possible null value, and I don't see a reason why you should send a null value to the query